cmake/SetupCharmAlgorithm.py

```diff
@@ -31,7 +31,8 @@ def create_interface_file(args):
               "Parallel::CProxy_ConstGlobalCache<\n" \
```

Comment:
Typos in commit message (Improve handling of options...):

* tuples -> tuple in third bullet point
* functionc
* Compoenents
* Singular-plural mismatch in bullet 6